### PR TITLE
Update openapi to use kube-openapi code

### LINKED ISSUE #51823: Cleanup openapi pkg
Addresses the comments left in https://github.com/kubernetes/kubernetes/pull/51762, including:
- add UT for gvk.go
- tweak the signature and interface a bit to make it better.

### PR DESCRIPTION
**Pull request by apelisse**
**What this PR does / why we need it**: OpenAPI code has moved to `github.com/kubernetes/kube-openapi`. Let's use that code as a dependency, since now it's duplicated.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #51823

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

No user visible changes. Just code moving around.

### CODE REVIEW

File: pkg/kubectl/cmd/util/openapi/openapi.go [hunk 1]
```diff
@@ -17,236 +17,108 @@ limitations under the License.
...
+// groupVersionKindExtensionKey is the key used to lookup the
+// GroupVersionKind value for an object definition from the
+// definition's "extensions" map.
+const groupVersionKindExtensionKey = "x-kubernetes-group-version-kind"
 
-// Path helps us keep track of type paths
-type Path struct {
-	parent *Path
-	key    string
+// Definitions is an implementation of `Resources`. It looks for
```
Comment (mengqiy) [inline pkg/kubectl/cmd/util/openapi/openapi.go]: Do you want to say `document is an implementation of Resources.`?

File: pkg/kubectl/cmd/util/openapi/openapi.go [hunk 2]
```diff
@@ -17,236 +17,108 @@ limitations under the License.
...
-// FieldPath appends a field name and creates a new path
-func (p *Path) FieldPath(field string) Path {
-	return Path{
-		parent: p,
-		key:    fmt.Sprintf(".%s", field),
+	resources := map[schema.GroupVersionKind]string{}
+	for _, modelName := range models.ListModels() {
+		model := models.LookupModel(modelName)
+		if model == nil {
+			panic("ListModels returns a model that can't be looked-up.")
```
Comment (mengqiy) [inline pkg/kubectl/cmd/util/openapi/openapi.go]: Why panic instead of error?

**Comment by mengqiy** [comment]
/assign @monopole 
for approving the change of moving code related to kubectl.

File: pkg/kubectl/cmd/util/openapi/openapi.go (hunk 1 above)
Comment (apelisse) [reply pkg/kubectl/cmd/util/openapi/openapi.go]: yes, thank you:-)

File: pkg/kubectl/cmd/util/openapi/openapi.go (hunk 2 above)
Comment (apelisse) [reply pkg/kubectl/cmd/util/openapi/openapi.go]: Because it's really really not supposed to happen.

**Comment by apelisse** [comment]
Fixed, PTAL!

**Pull request merged**